lib/ret_web/controllers/api/v1/media_controller.ex

```diff
@@ -11,20 +11,24 @@ defmodule RetWeb.Api.V1.MediaController do
   end
 
   def create(conn, %{
-        "media" =>
-          %Plug.Upload{filename: filename, content_type: "application/octet-stream"} = upload
+        "media" => %Plug.Upload{filename: filename, content_type: "application/octet-stream"} = upload,
+        "with_promotion_token" => with_promotion_token
```

Comment:
don't have a strong opinion, but maybe worth doing `promotion_mode: [always|with_token]` (leaving the option open for `never`)